### ISSUE TITLE
Add 'Stop to refesh button' on 'search online' on 'plugins

### ISSUE THREAD

**Issue by dateno1** [rejetto/hfs#1028]
**Is your feature request related to a problem? Please describe.**

Add button to 'stop auto refresh'

**Describe the solution you'd like**

It was refreshed again and again i wasn't do anything

**Describe alternatives you've considered**

It automatically refreshed again and again in short time (maybe 1~2min?)

It make hard to check list

Also item position on List is not fixed (example : 1,2,3,4,5 -> refreshed -> 1,2,5,4,3)

Maybe no one want to refresh it again on short period

I think need to Remove 'auto refresh' + Add 'Refresh button' or Add 'Stop Refresh button'

**Additional context**

**Comment by rejetto** [comment]
there's no "auto refresh" on that page,
and neither i ever saw any reload on it.
i just tried and waited for 10 minutes, nothing.
there may just be a problem on your side.
let me know if you find how to reproduce it.
try using your browser in incognito and see if the problem goes away. 

the order is not fixed, right, it is the one the server (github) provides.
that was my choice to avoid deciding what plugins to show first, so to avoid giving a "visibility advantage" of some type. Surely this is not very important.
I guess the order is an issue only because you have the reload issue. Normally the list is steady and the order is not annoying, i guess.

**Comment by dateno1** [comment]
Browser is not problem

![Image](https://github.com/user-attachments/assets/78491813-5373-495b-9954-9a24277e3098)

I found it was reproducted If i use reverse proxy

'127.0.0.1/~/admin' is OK

I used NGINX for reverse proxy

--------------------------------
nginx configuration

```
server {
	listen       80;
	listen       [::]:80;
	server_name  hfs.example.com;
	client_body_timeout 9999;

	charset utf-8;

	access_log  /var/log/nginx/access/hfs.example.com.log  combined;
	error_log   /var/log/nginx/error/hfs.example.com.log;

	autoindex off;

	return 301 https://hfs.example.com;

	include /usr/local/nginx/common/ERROR.conf;
}

server {
	listen       443 ssl;
	listen       [::]:443 ssl;
	listen       443 quic;
	listen       [::]:443 quic;
	http2 on;
	http3 on;
	server_name  hfs.example.com;
	client_body_timeout 9999;

	charset utf-8;

	access_log  /var/log/nginx/access/hfs.example.com.log  combined;
	error_log   /var/log/nginx/error/hfs.example.com.log;

	location / {
		proxy_redirect off;
		proxy_set_header Host $host:$server_port;
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		proxy_set_header X-Real-IP $remote_addr;
		proxy_set_header X-Forwarded-host $http_host;

		proxy_pass   https://192.168.100.16:443;
		
		proxy_http_version 1.1;

		valid_referers none blocked example.com *.example.com;
		if ($invalid_referer) {
			#return 403 break;
			rewrite ^(.*)$ /403.jpg break;
		}
		#limit_req      zone=one burst=30 nodelay;
	}

	autoindex off;

	#Set favicon.ico&Disable error on log
	location /favicon.ico {
		alias /usr/local/nginx/html/Favicon/HFS.ico;
		log_not_found off;
		access_log off;
	}

	include /usr/local/nginx/common/TLS.conf;
	include /usr/local/nginx/common/Policy.conf;
	include /usr/local/nginx/common/CSP.conf;
	include /usr/local/nginx/common/ERROR.conf;
}
```

--------------------
TLS.conf

```
ssl_dhparam  /usr/local/nginx/DH.pem;

ssl_session_timeout  30m;
ssl_session_cache    shared:SSL:30m;
ssl_session_tickets off;

ssl_buffer_size 32k;

ssl_protocols  TLSv1.2 TLSv1.3;

#Add http/3
#Enable Address Validation
quic_retry on;
#Enable Generic Segmentation Offloading
quic_gso on;

#can find TLS1.3 cipher list from https://wiki.openssl.org/index.php/TLS1.3
ssl_ciphers  EECDH+AESGCM:TLS-CHACHA20-POLY1305-SHA256:!RSA+AES:!aNULL:!MD5:!DSS:!DHE:!kEDH:HIGH:!eNULL:!EXPORT:!DES:!RC4:!PSK:!AECDH:!LOW:!SRP:!ADH:!COMPLEMENTOFDEFAULT:!TLS-AES-128-GCM-SHA256:!TLS-AES-128-CCM-SH256:!TLS-128-AES-CCM-8-SH256:!ECDHE-RSA-AES128-SHA256:!ECDHE-RSA-AES256-SHA384:!ECDHE-RSA-AES128-SHA:!ECDHE-RSA-AES256-SHA;

ssl_ecdh_curve X25519MLKEM768:secp521r1:secp384r1:X25519;
ssl_prefer_server_ciphers   on;

ssl_verify_depth 3;

add_header Strict-Transport-Security "max-age=31536000; includeSubDomains; preload" always;

ssl_trusted_certificate /usr/local/nginx/CA.pem;
ssl_certificate      /usr/local/nginx/Certificate.pem;
ssl_certificate_key  /usr/local/nginx/Certificate.key;
```

---------------------
Error.conf

```
# redirect server error pages to the static page /50x.html
#
error_page   400 401 402 405 406 407 408 409 410 411 412 413 414 415 416 417 495 496 497 501 502 504 505 506 507 /error.html;
location = /error.html {
	internal;
}
error_page   403 /403.jpg;
location = /403.jpg {
	internal;
}
error_page   404 /404.jpg;
location = /404.jpg {
	internal;
}
error_page   500 /500.jpg;
location = /500.jpg {
	internal;
}
error_page   503 /503.jpg;
location = /503.jpg {
	internal;
}
```

------------------------------
CSP.conf

```
set $CSP "";
set $CSP "${CSP}default-src 'self' https://example.com https://*.example.com wss://example.com wss://*.example.com;";
set $CSP "${CSP}base-uri 'self' https://example.com https://*.example.com wss://example.com wss://*.example.com;";
set $CSP "${CSP}child-src 'self' https://example.com https://*.example.com wss://example.com wss://*.example.com;";
set $CSP "${CSP}connect-src 'self' https://example.com https://*.example.com wss://example.com wss://*.example.com https://api-cdn.embed.ly;";
set $CSP "${CSP}font-src 'self' data: https://example.com https://*.example.com wss://example.com wss://*.example.com;";
set $CSP "${CSP}form-action 'self' https://example.com https://*.example.com wss://example.com wss://*.example.com https://*.googlevideo.com https://*.youtube.com;";
set $CSP "${CSP}frame-ancestors 'self' https://example.com https://*.example.com wss://example.com wss://*.example.com;";
set $CSP "${CSP}frame-src 'self' https://example.com https://*.example.com wss://example.com wss://*.example.com;";
set $CSP "${CSP}img-src 'self' data: blob: mediastream: filesystem: https: https://example.com https://*.example.com wss://example.com wss://*.example.com"; 
set $CSP "${CSP}manifest-src 'self' data: https://example.com https://*.example.com wss://example.com wss://*.example.com;";
set $CSP "${CSP}media-src 'self' data: blob: mediastream: filesystem: https://example.com https://*.example.com wss://example.com wss://*.example.com;";
set $CSP "${CSP}object-src 'self' https://example.com https://*.example.com wss://example.com wss://*.example.com;";

# If you need Java UnComment it
# set $CSP "${CSP}plugin-types application/x-java-applet;";
# You can't Remove 'unsafe-eval'&'unsafe-inline' for hfs(node) (Not Work)
set $CSP "${CSP}script-src 'self' 'unsafe-eval' 'unsafe-inline' 'wasm-unsafe-eval' https://example.com https://*.example.com wss://example.com wss://*.example.com https://ajax.googleapis.com https://*.embedly.com https://cdnjs.cloudflare.com https://postimgs.org https://*.embed.ly https://unpkg.com;";

set $CSP "${CSP}script-src-attr 'self' 'unsafe-inline' https://example.com https://*.example.com wss://example.com wss://*.example.com;";
set $CSP "${CSP}script-src-elem 'self' 'unsafe-inline' https://example.com https://*.example.com wss://example.com wss://*.example.com https://ajax.googleapis.com https://*.embedly.com https://cdnjs.cloudflare.com https://postimgs.org https://*.embed.ly https://unpkg.com;";


set $CSP "${CSP}style-src 'self' 'unsafe-inline' https://example.com https://*.example.com wss://example.com wss://*.example.com  https://fonts.googleapis.com https://postimgs.org https://*.embedly.com https://*.embed.ly https://unpkg.com;";

set $CSP "${CSP}worker-src 'self' data: blob: filesystem: https://example.com https://*.example.com wss://example.com wss://*.example.com;";

# set $CSP "${CSP}require-trusted-types-for 'script';";

set $CSP "${CSP}upgrade-insecure-requests; block-all-mixed-content;";
add_header Content-Security-Policy $CSP;
```

--------------------------
Policy.conf

```
add_header Permissions-Policy "accelerometer=(), ambient-light-sensor=(), autoplay=(self), battery=(self), camera=(self), cross-origin-isolated=(self), display-capture=*, document-domain=(self), encrypted-media=(self), fullscreen=(self), geolocation=(self), gyroscope=(), layout-animations=(self), legacy-image-formats=*, magnetometer=(), microphone=(self), midi=(), oversized-images=*, payment=(), picture-in-picture=(self), speaker=(self), sync-xhr=(self), unoptimized-images=*, unsized-media=*, usb=(self), vr=(self), web-share=(self), clipboard-read=(self), clipboard-write=(self), gamepad=*, speaker-selection=*" always;

fastcgi_hide_header X-Powered-By;
proxy_hide_header X-Powered-By;
proxy_hide_header Server;

add_header Cache-Control no-store;
add_header Pragma no-cache;
add_header Referrer-Policy strict-origin always;
add_header X-Content-Type-Options "nosniff";
add_header X-Frame-Options "SAMEORIGIN" always;
add_header X-Permitted-Cross-Domain-Policies none;
add_header X-XSS-Protection "1; mode=block";

#Add http3
add_header Alt-Svc 'h3=":443"; ma=86400';

proxy_cookie_path ~(.*) "$1; SameSite=strict; secure; httponly";
```

------------------------------------------------------

Domain is masked by example.com

**Comment by rejetto** [comment]
i can confirm that using nginx as proxy will cause this problem.
it applies to a lot of pages of the admin-panel, like most of the pages that have a list.
i also verified that it's nginx actively closing the connection, not hfs.
solution seems to be adding this line to nginx config:
proxy_read_timeout 1h;

it solved for me.
i just updated the wiki page https://github.com/rejetto/hfs/wiki/Reverse-proxy

**Comment by dateno1** [comment]
Thanks for help

I will test it
